hrc_speech_prediction/tests/test_context_model.py

```diff
@@ -0,0 +1,86 @@
+from unittest import TestCase
+
+from hrc_speech_prediction.data import ALL_ACTIONS
+from hrc_speech_prediction.context_model import ContextTreeModel, Node
+
+
+class TestNode(TestCase):
+
+    def setUp(self):
+        self.empty_context = []
+        self.context = ['a','b','c']
+        self.contexts = [['a','b','c'],['d','e','f'],['g','h','i'],['a','g','e']]
+
+    def test_empty_has_zero_n_children(self):
+        x = Node()
+        self.assertEqual(x.n_children, 0)
+
+    def test_nonempty_has_n_children(self):
+        x = Node()
+        x.get_or_add_node(self.empty_context)
+        self.assertEqual(x.n_children, 0)
+      
+        x.get_or_add_node(self.context)
+        self.assertEqual(x.n_children,1)
+
+        x.get_or_add_node(self.context)
+        self.assertEqual(x.n_children, 1)
+
+        x.get_or_add_node(['b','c'])
+        self.assertEqual(x.n_children, 2)
+
+        self.assertEqual(x.n_children, len(x._children))
+
+    def test_add_new_node_adds_child(self):
+        x = Node()
+        x.get_or_add_node(self.context)
+
+        self.assertTrue(self.context[0] in x._children.keys())
+        self.assertTrue(self.context[1] in x._children[self.context[0]]._children.keys())
+        self.assertTrue(self.context[2] in x._children[self.context[0]]._children[self.context[1]]._children.keys())
+
+    def test_add_branch_increments_count(self):
+        x = Node()
+        self.assertEqual(x._count,0)
+        branch = x.add_branch(self.context)
+        self.assertEqual(branch._count, 1)
+        branch = x.add_branch(self.context)
+        self.assertEqual(branch._count, 2)
+
+        self.assertEqual(x._children['a']._children['b']._count, 0)
+        branch = x.add_branch(['a','b'])
+        self.assertEqual(branch._count, 1)
+
+class TestContextTreeModel(TestCase):
+    def setUp(self):
```

Comment:
I know that this was the case also in the previous version, but for uniformity I would change the name of this function from `setUp` to `set_up`

Reply:
That would indeed make more sense. Unfortunately, this is the name of the method that the `unittest` library is calling before running each test so we have to stick with it… And, yes, Python standard libraries are often very bad at abiding by the language's own standards!